### ISSUE TITLE
Modulesd crashes when bad ciscat wodle configuration

### ISSUE THREAD

**Issue by manuasir** [wazuh/wazuh#1828]
## Bug report

**OS**
_Ubuntu16_

**Wazuh version**
_3.7.0-3717_

**Install type**
_agent_

**Install method**
_packages_

**Log sample**

```
2018/11/07 16:33:40 wazuh-modulesd: ERROR: Invalid command: './CIS-CAT.sh -a -b /tmp/ciscat/benchmarks/CIS_Ubuntu_Linux_16.04_LTS_Benchmark_v1.0.0-xccdf.xml -p xccdf_org.cisecurity.benchmarks_profile_Level_2_-_Server -r /var/ossec/tmp -rn ciscat-report -x -t -n -y': (2) No such file or directory
2018/11/07 16:33:40 wazuh-modulesd:ciscat: CRITICAL: Internal calling. Exiting...
2018/11/07 16:33:40 wazuh-modulesd:ciscat: INFO: Module finished.
2018/11/07 16:33:40 wazuh-modulesd:syscollector: INFO: Module finished.
2018/11/07 16:33:40 wazuh-modulesd: ERROR: Couldn't delete PID file.

```

Bug description
----------------
When I try to run the `ciscat` wodle without execution permissions to the `CIS-CAT.sh` script, modulesd crashes and I get the above `2018/11/07 16:33:40 wazuh-modulesd: ERROR: Couldn't delete PID file` log. Additionally, when I set the execution permission to the script and restart the agent, I realized that I got two processes of `modulesd`:
```
Every 2.0s: ps -A |grep wazuh                     Wed Nov  7 16:37:25 2018

22598 ?        00:00:00 wazuh-modulesd
22621 ?        00:00:00 wazuh-modulesd
```

Thanks

**Comment by bah07** [comment]
The problem is that each thread of the _wazuh-modulesd_ tries to delete the file `wazuh-modulesd-<PID>.pid` when it closes unexpectedly. This file only exists for the PID of the _Modulesd_ parent process. The error is caused by trying to delete a file that does not exist.
We must control in the function `wm_cleanup()` that the process that is closing deletes this file only if it is the father.

**Comment by vikman90** [comment]
This issue was fixed in PR #1836.
